demo/c/koala_demo_mic.c

```diff
@@ -0,0 +1,424 @@
+    if (recorder_status != PV_RECORDER_STATUS_SUCCESS) {
+        fprintf(stderr, "Failed to initialize device with %s.\n", pv_recorder_status_to_string(recorder_status));
+        exit(EXIT_FAILURE);
+    }
+
+    const char *selected_device = pv_recorder_get_selected_device(recorder);
+    fprintf(stdout, "Selected device: %s.\n", selected_device);
+
+    fprintf(stdout, "Start recording (press Ctrl+C to stop)...\n");
+    recorder_status = pv_recorder_start(recorder);
+    if (recorder_status != PV_RECORDER_STATUS_SUCCESS) {
+        fprintf(stderr, "Failed to start device with %s.\n", pv_recorder_status_to_string(recorder_status));
+        exit(EXIT_FAILURE);
+    }
+
+    int16_t *pcm = (int16_t *) malloc(frame_length * sizeof(int16_t));
+    if (!pcm) {
+        fprintf(stderr, "Failed to allocate pcm memory.\n");
+        exit(EXIT_FAILURE);
+    }
+
+    int16_t *enhanced_pcm = (int16_t *) malloc(frame_length * sizeof(int16_t));
+    if (!enhanced_pcm) {
+        fprintf(stderr, "Failed to allocate enhanced_pcm memory.\n");
+        exit(EXIT_FAILURE);
+    }
+
+    while (!is_interrupted) {
+        recorder_status = pv_recorder_read(recorder, pcm);
+        if (recorder_status != PV_RECORDER_STATUS_SUCCESS) {
+            fprintf(stderr, "Failed to read with %s.\n", pv_recorder_status_to_string(recorder_status));
+            exit(EXIT_FAILURE);
+        }
+
+        koala_status = pv_koala_process_func(koala, pcm, enhanced_pcm);
+        if (koala_status != PV_STATUS_SUCCESS) {
+            fprintf(stderr, "'pv_koala_process' failed with '%s'\n", pv_status_to_string_func(koala_status));
+            exit(EXIT_FAILURE);
+        }
+
+        if ((int32_t) drwav_write_pcm_frames(&output_file, frame_length, enhanced_pcm) != frame_length) {
+            fprintf(stderr, "Failed to write to wav file.\n");
+            exit(EXIT_FAILURE);
+        }
+
+        if (reference_path) {
+            if ((int32_t) drwav_write_pcm_frames(&reference_file, frame_length, pcm) != frame_length) {
+                fprintf(stderr, "Failed to write to reference wav file.\n");
+                exit(EXIT_FAILURE);
+            }
+        }
+
+        print_vu_meter(pcm, frame_length);
+    }
+    fprintf(stdout, "\n");
+
+    recorder_status = pv_recorder_stop(recorder);
+    if (recorder_status != PV_RECORDER_STATUS_SUCCESS) {
+        fprintf(stderr, "Failed to stop device with %s.\n", pv_recorder_status_to_string(recorder_status));
+        exit(1);
+    }
+
+    free(pcm);
+    free(enhanced_pcm);
+    drwav_uninit(&output_file);
+    if (reference_path) {
+        drwav_uninit(&reference_file);
+    }
+    pv_recorder_delete(recorder);
+    pv_koala_delete_func(koala);
+    close_dl(koala_library);
+
+    return EXIT_SUCCESS;
+}
+
+int main(int argc, char *argv[]) {
+
+#if defined(_WIN32) || defined(_WIN64)
+
+#define UTF8_COMPOSITION_FLAG (0)
+#define NULL_TERMINATED (-1)
```

Comment:
These need to be moved to the top as the other "multibyte" sections refer to them